distribution/lib/Standard/Base/0.0.0-dev/src/Data.enso

```diff
@@ -92,6 +94,59 @@ read path=(Missing_Argument.throw "path") format=Auto_Detect (on_problems : Prob
         if file_obj.is_directory then Error.throw (Illegal_Argument.Error "Cannot `read` a directory, use `Data.list`.") else
             file_obj.read format on_problems
 
+## ALIAS load, open
+   GROUP Input
+   ICON data_input
+   Reads a a list of files into Enso.
+
+   Arguments:
+   - paths: A list of files to load. It can be a Vector, Column or Table of
+     files, paths or URIs to fetch. If a Table is provided, it must either
+     contain a single column or a column called `path` (case insensitive).
+   - format: A `File_Format` object used to read files into memory.
+     If `Auto_Detect` is specified; each file determines the specific
+     type and configures it appropriately. If there is no matching type then
+     a `File_Error.Unsupported_Type` error is returned.
+   - return: Specifies the shape of the data to return.
+   - on_problems: Specifies the behavior when a problem occurs during the
+     function.
+     By default, if one of the files fails to load, a warning is issued and the
+     entry for that file becomes `Nothing`, but the operation proceeds.
+     If set to `Report_Error`, the operation fails with a dataflow error on the
+     first failing file.
+     If set to `Ignore`, the operation proceeds without errors or warnings,
+     replacing files that fail to load with `Nothing`.
+
+   ! Request Caching
+
+     Responses to HTTP data requests are cached, and additional requests for the
+     same resources will use the cache, saving a round-trip call to the remote
+     server. Two resources are considered the same if the URIs and request
+     headers are the same. Header order does not affect sameness.
+
+     The cache respects the "max-age" and "Age" response headers; see
+     `Data.fetch` for more details.
+
+     The cached values are retained as long as the project remains open. Closing
+     a project will clear the cache.
+
+   > Example
+     Read all CSV files from a directory into a single merged table.
+
+         from Standard.Table import all
+         import Standard.Examples
+
+         files = Data.list name_filter="*.csv"
+         example_csv_dir_to_table = Data.read_many files
+@paths (Vector_Editor item_editor=Text_Input item_default='""')
+@format File_Format.default_widget
+read_many : Many_Files_List -> File_Format -> Return_As -> Problem_Behavior -> Any ! File_Error
```

Comment:
Why don't we use inline type signatures - e.g. why don't we enforce the types during runtime?

Reply:
What is `(paths : Many_Files_List = Missing_Argument.throw "paths")` if not the inline type signature?

For `Return_As` and `File_Format` we do `.resolve` tricks that allow us to have "dynamic" auto-scoped constructors - i.e. allows to write e.g. `..Excel` for a type that is not even defined in `Base` - it is resolved because `FileFormatSPI` knows all available types and calls `resolve` on them hoping some will match. This is a special trick used only in a few places to provide comfortable UX to users and it is incompatible with inline type checks.